Florida_Analysis_Py_File.py

```diff
@@ -0,0 +1,534 @@
+# calculating yearwise drug related deaths results for Tennessee
+tennessee_results_yearwise = (
+    other_states_deaths_copy[other_states_deaths_copy["State"] == "Tennessee"]
+    .groupby("Year")["Overdose_Per_100K"]
+    .mean()
+)
+tennessee_results_yearwise
+# calculating and displaying yearwise results for drug related deaths in neighboring states around Florida (combined)
+
+other_states_results_combined_yearwise = other_states_deaths_copy.groupby("Year")[
+    "Overdose_Per_100K"
+].mean()
+other_states_results_combined_yearwise
+
+# these yearwise results have been calculated while doing the pre-post analysis of drug overdose deaths in Florida
+florida_results_yearwise
+florida_summary = pd.DataFrame(florida_result.describe()["Overdose_Per_100K"]).rename(
+    columns={"Overdose_Per_100K": "Overdoses per 100k Residents - Florida"}
+)
+other_summary = pd.DataFrame(
+    other_states_result.describe()["Overdose_Per_100K"]
+).rename(columns={"Overdose_Per_100K": "Overdoses Per 100K Residents - Control States"})
+summary_statistics = pd.concat([florida_summary, other_summary], axis=1)
+summary_statistics
+other_states_result_copy = other_states_result.copy()
+
+other_states_result_copy["Year Relative To Policy"] = other_states_result_copy[
+    "Year"
+].apply(lambda x: scale_years(x))
+
+# double check no nulls in "year relative to policy"
+
+assert other_states_result_copy["Year Relative To Policy"].isnull().sum() == 0
+
+# doing this in case the float == int comparison causes issues
+
+other_states_before = other_states_result_copy[other_states_result_copy["Year"] < 2010]
+other_states_after = other_states_result_copy[other_states_result_copy["Year"] > 2010]
+
+regressor_other_states_before = LinearRegression()
+regressor_other_states_after = LinearRegression()
+
+
+x_other_before = np.array(other_states_before["Year Relative To Policy"]).reshape(-1, 1)
+y_other_before = np.array(other_states_before["Overdose_Per_100K"]).reshape(-1, 1)
+
+x_other_after = np.array(other_states_after["Year Relative To Policy"]).reshape(-1, 1)
+y_other_after = np.array(other_states_after["Overdose_Per_100K"]).reshape(-1, 1)
+
+
+regressor_other_states_before.fit(x_other_before, y_other_before)
+regressor_other_states_after.fit(x_other_after, y_other_after)
+
+
+y_pred_other_before = regressor_other_states_before.predict(x_other_before)
+y_pred_other_after = regressor_other_states_after.predict(x_other_after)
+plt.xlim(-3, 3)
+# plt.ylim(0, 500)
+
+plt.xlabel("Year Relative To Policy Change Year (2010)")
+plt.ylabel("Deaths due to Drug Overdose (Per 100K)")
+plt.title("Difference in Difference Analysis of Drug Overdose Deaths In Florida")
+
+plt.plot(x_other_before, y_pred_other_before, color="green", label="Other States")
+plt.plot(x_other_after, y_pred_other_after, color="green", label="Other States")
+
+plt.plot(x_before, y_pred_before, color="orange", label="Florida")
+plt.plot(x_after, y_pred_after, color="orange", label="Florida")
```

Comment:
Hey Annapurna, it's so nice seeing you add labels here. Maybe you can also embed labels in the previous part?

Reply:
referring to my last comment

Reply:
Thanks Zhonglin. I will surely add the labels in the previous part.